app/src/main/java/com/example/keniphhomemaintenance/dwellings/DwellingsScreenView.kt

```diff
@@ -0,0 +1,226 @@
+package com.example.keniphhomemaintenance.dwellings
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.KeyboardArrowDown
+import androidx.compose.material.icons.filled.KeyboardArrowUp
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.unit.toSize
+import androidx.compose.ui.window.Dialog
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.navigation.NavController
+
+
+val addDwellingDialog = mutableStateOf(false)
+
+@ExperimentalComposeUiApi
+@Composable
+fun DwellingsScreenView(navController: NavController) {
+
+    ConstraintLayout {
+
+        val (lazyColumn, scaffold) = createRefs()
+
+        val listOfDwellings = dwellingsViewModel.dwellingsList.value
```

Comment:
Think I saw above somewhere that the UI wasn't updating properly when you added things to the list. It's because you're getting the value here, not using MutableState as it's intended. See https://stackoverflow.com/a/67252955/3875775

Reply:
Ope maybe not, .value works different on MutableStates than on LiveDatas, possibly my bad